bundle/python_bundle_workflow/scripts/train.py

```diff
@@ -0,0 +1,297 @@
+        return Compose(
+            [
+                LoadImaged(keys=["image", "label"]),
+                AsChannelFirstd(keys=["image", "label"], channel_dim=-1),
+                ScaleIntensityd(keys="image"),
+                RandCropByPosNegLabeld(
+                    keys=["image", "label"],
+                    label_key="label",
+                    spatial_size=[96, 96, 96],
+                    pos=1,
+                    neg=1,
+                    num_samples=4,
+                ),
+                RandRotate90d(keys=["image", "label"], prob=0.5, spatial_axes=[0, 2]),
+            ]
+        )
+
+    @property
+    def _train_postprocessing(self):
+        return Compose(
+            [
+                Activationsd(keys="pred", sigmoid=True),
+                AsDiscreted(keys="pred", threshold=0.5),
+                KeepLargestConnectedComponentd(keys="pred", applied_labels=[1]),
+            ]
+        )
+
+    @property
+    def _train_key_metric(self):
+        return ({"train_acc": Accuracy(output_transform=from_engine(["pred", "label"]))},)
+
+    @property
+    def _evaluator(self):
+        return SupervisedEvaluator(
+            device=self.device,
+            val_data_loader=DataLoader(self.val_dataset, batch_size=1, num_workers=4),
+            network=self.network,
+            inferer=self.val_inferer,
+            postprocessing=self.val_postprocessing,
+            key_val_metric=self.val_key_metric,
+            additional_metrics={"val_acc": Accuracy(output_transform=from_engine(["pred", "label"]))},
+            val_handlers=self.val_handlers,
+            amp=True,
+        )
+
+    @property
+    def _val_interval(self):
+        return 2
+
+    @property
+    def _val_handlers(self):
+        return [
+            # use the logger "train_log" defined at the beginning of this program
+            StatsHandler(name="train_log", output_transform=lambda x: None),
+            CheckpointSaver(
+                save_dir=self.bundle_root + "/models/",
+                save_dict={"net": self.network},
+                save_key_metric=True,
+                key_metric_filename="model.pt",
+            ),
+        ]
+
+    @property
+    def _val_dataset(self):
+        return CacheDataset(data=self.val_dataset_data, transform=self.val_preprocessing, cache_rate=1.0)
+
+    @property
+    def _val_dataset_data(self):
+        images = sorted(glob(os.path.join(self.dataset_dir, "img*.nii.gz")))
+        segs = sorted(glob(os.path.join(self.dataset_dir, "seg*.nii.gz")))
+        return [{"image": img, "label": seg} for img, seg in zip(images[-20:], segs[-20:])]
+
+    @property
+    def _val_inferer(self):
+        return SlidingWindowInferer(roi_size=(96, 96, 96), sw_batch_size=4, overlap=0.5)
+
+    @property
+    def _val_preprocessing(self):
+        return Compose(
+            [
+                LoadImaged(keys=["image", "label"]),
+                AsChannelFirstd(keys=["image", "label"], channel_dim=-1),
```

Comment:
Also here.

Reply:
Sure, updated in the latest commit.

Thanks.